src/rpc/chain.ts

```diff
@@ -999,6 +999,170 @@ export class ChainRpc {
         });
     }
+
+    /**
+     * Gets the number of shards, at the state of given blockNumber
+     * @param blockNumber A block number.
+     * @returns A number of shards
+     */
+    public getNumberOfShards(blockNumber?: number): Promise<number> {
+        if (blockNumber !== undefined && !isNonNegativeInterger(blockNumber)) {
+            throw Error(
+                `Expected the second argument of getRegularKey to be a number but found ${blockNumber}`
+            );
+        }
+
+        return new Promise((resolve, reject) => {
+            this.rpc
+                .sendRpcRequest("chain_getNumberOfShards", [blockNumber])
+                .then(result => {
+                    if (result === null || typeof result === "number") {
+                        resolve(result);
+                    } else {
+                        reject(
+                            Error(
+                                `Expected chain_getNumberOfShards to return either null or number, but it returned ${result}`
+                            )
+                        );
+                    }
+                })
+                .catch(reject);
+        });
+    }
+
+    /**
+     * Gets the platform account in the genesis block
+     * @returns The platform addresses in the genesis block.
+     */
+    public getGenesisAccounts(): Promise<PlatformAddress[]> {
+        return new Promise((resolve, reject) => {
+            this.rpc
+                .sendRpcRequest("chain_getGenesisAccounts", [])
+                .then(result => {
+                    try {
+                        resolve(
+                            (result as string[]).map(str =>
+                                PlatformAddress.ensure(str)
+                            )
+                        );
+                    } catch (e) {
+                        reject(
+                            Error(
+                                `Expected chain_getGenesisAccounts to return an array of PlatformAddress, but an error occurred: ${e.toString()}`
+                            )
+                        );
+                    }
+                })
+                .catch(reject);
+        });
+    }
+
+    /**
+     * Gets the root of shard, at the state of the given blockNumber.
+     * @param shardId A shard Id.
+     * @param blockNumber A block number.
+     * @returns The hash of root of shard.
+     */
+    public getShardRoot(
+        shardId: number,
+        blockNumber?: number
+    ): Promise<H256 | null> {
+        if (!isShardIdValue(shardId)) {
+            throw Error(
+                `Expected the second argument of getAssetSchemeByTracker to be a shard ID value but found ${shardId}`
+            );
+        }
+        if (blockNumber !== undefined && !isNonNegativeInterger(blockNumber)) {
+            throw Error(
+                `Expected the third argument of getAssetSchemeByTracker to be non-negative integer but found ${blockNumber}`
```

Comment:
Expected the third argument of getAssetSchemeByTracker to be a non-negative integer but found